sources/awscli-2.patch

```diff
@@ -0,0 +1,48 @@
+diff -urN aws-cli-2.4.12-orig/awscli/customizations/wizard/app.py aws-cli-2.4.12/awscli/customizations/wizard/app.py
+--- aws-cli-2.4.12-orig/awscli/customizations/wizard/app.py	2022-01-18 22:00:18.000000000 +0000
++++ aws-cli-2.4.12/awscli/customizations/wizard/app.py	2022-01-20 22:02:20.045784837 +0000
+@@ -12,10 +12,10 @@
+ # language governing permissions and limitations under the License.
+ import json
+ import os
++from asyncio import get_event_loop
+ from collections.abc import MutableMapping
+ 
+ from prompt_toolkit.application import Application
+-from prompt_toolkit.eventloop import get_event_loop, run_until_complete
+ 
+ from awscli.customizations.wizard import core
+ from awscli.customizations.wizard.ui.style import get_default_style
+@@ -68,7 +68,7 @@
+         loop.set_exception_handler(self._handle_exception)
+         try:
+             f = self.run_async(pre_run=pre_run)
+-            run_until_complete(f)
++            loop.run_until_complete(f)
+             return f.result()
+         finally:
+             loop.set_exception_handler(previous_exc_handler)
+diff -urN aws-cli-2.4.12-orig/setup.cfg aws-cli-2.4.12/setup.cfg
+--- aws-cli-2.4.12-orig/setup.cfg	2022-01-18 22:00:18.000000000 +0000
++++ aws-cli-2.4.12/setup.cfg	2022-01-20 21:49:45.335819606 +0000
+@@ -28,14 +28,13 @@
+ python_requires = >=3.7
+ include_package_data = True
+ install_requires =
+-    colorama>=0.2.5,<0.4.4
```

Comment:
for the awscli v1 we relax the dependencies to prevent running into complications with independent updates for python packages as much as possible. We may just need a script to relax the dependencies to fit the versions that are in Fedora when they advance.

Reply:
Yeah I don't think I have a strong opinion on what the version dependency ranges are in the Fedora patch. I think in v2, we plan to continue to pin to a minor version but we just got to get better at updating the range when new versions come out so ideally we would not even need the patch.